build-support/bin/generate_github_workflows.py

```diff
@@ -660,91 +589,69 @@ def macos11_x86_64_jobs(python_versions: list[str], *, cron: bool) -> Jobs:
             ],
         },
     }
-    if not cron:
-        jobs.update(
-            {
-                "build_wheels_macos11_x86_64": {
-                    "name": f"Build wheels and fs_util ({helper.platform_name()})",
-                    "runs-on": helper.runs_on(),
-                    "timeout-minutes": 120,
-                    **helper.build_wheels_common,
-                    "steps": [
-                        *checkout(),
-                        setup_toolchain_auth(),
-                        expose_all_pythons(),
-                        # NB: We only cache Rust, but not `native_engine.so` and the Pants
-                        # virtualenv. This is because we must build both these things with
-                        # multiple Python versions, whereas that caching assumes only one primary
-                        # Python version (marked via matrix.strategy).
-                        *helper.rust_caches(),
-                        *helper.build_steps(),
-                        helper.upload_log_artifacts(name="wheels"),
-                        deploy_to_s3(),
-                    ],
-                },
-            }
-        )
     return jobs
 
 
-def macos_10_15_x86_64_jobs(python_versions: list[str]) -> Jobs:
-    helper = Helper(Platform.MACOS10_15_X86_64)
+def build_wheels_job(platform: Platform, python_versions: list[str]) -> Jobs:
+    helper = Helper(platform)
+    if platform == Platform.LINUX_X86_64:
+        # For manylinux compatibility, we build wheels in a container rather than directly
+        # on the Ubuntu runner. As a result, we have custom steps here to check out
+        # the code, install rustup and expose Pythons.
+        # TODO: Run tests etc. using this container image, so we can reuse the rust engine build?
```

Comment:
Tests run in debug mode by default, so they won't share any compilation with production/`release` wheel builds. The only time you would have overlap would be in PR runs that affected wheel builds... probably not worth it to try and share between tests and release builds.

Reply:
Makes sense, will adjust the comment

Reply:
But we could still profitably cache cargo etc presumably?

Reply:
Possibly, yea.